buildSrc/src/main/kotlin/mockito.java-library-conventions.gradle.kts

```
@@ -0,0 +1,40 @@
+plugins {
+    `java-library`
+    id("mockito.java-conventions")
+    id("mockito.test-conventions")
+    id("mockito.test-retry-conventions")
+    id("mockito.test-launcher-conventions")
+    id("net.ltgt.errorprone")
+}
+
+afterEvaluate {
+    if (!base.archivesName.get().startsWith("mockito-")) {
+        // TODO throw GradleException("Project module should be prefixed with `mockito-` : ${project.projectDir})")
+        logger.warn("Project module should be prefixed with `mockito-` : ${project.projectDir})")
+        base.archivesName = "mockito-" + project.name
+    }
+}
```

I don't think the `afterEvaluate` is required here and we should use the `project.name` instead to have the correct logic. Then, we can also immediately throw and enforce this invariant.

```suggestion
if (!project.getName().startsWith("mockito-")) {
    throw new GradleException("Project module should be prefixed with `mockito-` : ${project.projectDir})")
}
```
```

---

Good catch! I edited your suggestion to use the kotlin property access `.name` instead of `.getName()`